src/cider/exec/processor/StatefulProcessor.cpp

```diff
@@ -25,15 +25,26 @@
 
 namespace cider::exec::processor {
 
+StatefulProcessor::StatefulProcessor(const plan::SubstraitPlanPtr& plan,
+                                     const BatchProcessorContextPtr& context)
+    : DefaultBatchProcessor(plan, context) {
+  is_groupby_ = plan->isGroupingAggregateRel();
+}
+
 void StatefulProcessor::getResult(struct ArrowArray& array, struct ArrowSchema& schema) {
-  if (!no_more_batch_ || BatchProcessorState::kFinished == state_) {
-    input_arrow_array_ = nullptr;
+  if (!no_more_batch_ || !has_result_) {
```

Comment:
When will this happen? Shouldn't frontend be blocked until `BatchProcessorState::kFinished`?